src/drv_pwm.c

```diff
@@ -187,19 +189,19 @@ static void pwmOCConfig(TIM_TypeDef *tim, uint8_t channel, uint16_t value)
     switch (channel) {
         case TIM_Channel_1:
             TIM_OC1Init(tim, &TIM_OCInitStructure);
-            TIM_OC1PreloadConfig(tim, TIM_OCPreload_Enable);
+            TIM_OC1PreloadConfig(tim, TIM_OCPreload_Disable);
             break;
         case TIM_Channel_2:
             TIM_OC2Init(tim, &TIM_OCInitStructure);
-            TIM_OC2PreloadConfig(tim, TIM_OCPreload_Enable);
+            TIM_OC2PreloadConfig(tim, TIM_OCPreload_Disable);
             break;
         case TIM_Channel_3:
             TIM_OC3Init(tim, &TIM_OCInitStructure);
-            TIM_OC3PreloadConfig(tim, TIM_OCPreload_Enable);
+            TIM_OC3PreloadConfig(tim, TIM_OCPreload_Disable);
             break;
         case TIM_Channel_4:
             TIM_OC4Init(tim, &TIM_OCInitStructure);
-            TIM_OC4PreloadConfig(tim, TIM_OCPreload_Enable);
+            TIM_OC4PreloadConfig(tim, TIM_OCPreload_Disable);

```

Comment:
you are right, in rare cases, it could cause the output wrong values.
@fiendie will update the pr